pageserver/src/tenant/timeline.rs

```diff
@@ -4416,26 +4437,17 @@ impl Timeline {
     ///
     /// The 'pitr' duration is used to calculate a 'pitr_cutoff', which can be used to determine
     /// whether a record is needed for PITR.
-    ///
-    /// NOTE: This function holds a short-lived lock to protect the 'gc_info'
-    /// field, so that the three values passed as argument are stored
-    /// atomically. But the caller is responsible for ensuring that no new
-    /// branches are created that would need to be included in 'retain_lsns',
-    /// for example. The caller should hold `Tenant::gc_cs` lock to ensure
-    /// that.
-    ///
     #[instrument(skip_all, fields(timeline_id=%self.timeline_id))]
```

Comment:
You didn't update the full doc comment. First paragraph at the top still says "update information", which is wrong, it just finds it now, but doesn't update anything.
Please move that paragraph to the `update_gc_info` caller and find a new first paragraph.

Reply:
This removed paragraph is meaningless now because all of the gc_info modification happens in the same scope of holding `Tenant::gc_cs`.

I'll clean up the doc comment otherwise.

Reply:
Well, the `Tenant::gc_cs` is not required for the pitr search, it was only there because the function used to do:
1. find pitr_cutoff
2. do the in place update including retain_lsns

Now as it only does (1) and not the (2) it is not related.

Reply:
[9207f15](https://github.com/neondatabase/neon/pull/7584/commits/9207f15f2901715aee55f6d0524a50347980d899)